prow/cluster/velodrome/velodrome_service.yaml

```diff
@@ -5,6 +5,20 @@ metadata:
     app.kubernetes.io/part-of: velodrome
   name: velodrome
   namespace: velodrome
+spec:
+  type: NodePort
```

Comment:
Why are both velodrome and velodrome-external nodeports on port 80?

Reply:
Because local proxy is running on localhost:80 and external veldorome on 130.211.224.127:80

Reply:
The proxy is running in a container right? Why do we need to expose the host's port 80?

Reply:
For ingress.

Reply:
I thought the whole point here was to ensure all traffic from the internet to velodrome is https? That doesn't work when you're exposing it.

I'd expect ClusterIP

Reply:
ClusterIP does not work for GKE load balancer.

Reply:
You're using nginx, not the GKE load balancer?

Reply:
The whole point here is to route all traffic through nginx so we ensure it is https.

Therefore we explicitly want all traffic to the velodrome deployment to be internal, aka ClusterIP

Reply:
We are using a GKE load balancer like we use everywhere else for Ingress not the nginx load balancer. Nginx here is just a reverse proxy in a container.

Reply:
What is the desired packet flow? What is the purpose of nginx and how does it work?

Reply:
Exactly what you said:
User -> Internet -> Nginx -> Velodrome

Velodrome is deployed in an external cluster, so the internal service (nginx) does the http->https redirect then proxies traffic to the external one. Also, AFAIK (and have reconed) GCE firewalls blocks NodePort port access unless explicitly whitelisted on the node.

Reply:
* You cannot have two service using the same NodePort
* Since we're configuring velodrome to only accept internal requests from nginx, this should be ClusterIP

Reply:
> You cannot have two service using the same NodePort

The second service maps to an external source defined in the endpoint; there is not a conflict. see: https://kubernetes.io/docs/concepts/services-networking/service/#services-without-selectors

> Since we're configuring velodrome to only accept internal requests from nginx, this should be ClusterIP

[GKE load balancers with ingress](https://cloud.google.com/kubernetes-engine/docs/tutorials/http-balancer) **require** `NodePort` services. Same goes for every other service we expose via an ingress  currently deployed (e.g. Deck, Hook, etc.).